dataclasses_json/cfg.py

```diff
@@ -1,10 +1,23 @@
 import functools
-from typing import (Callable, Dict, Optional, Union)
+from typing import (Any, Callable, Dict, NamedTuple, Optional, Union)
 
 from marshmallow.fields import Field as MarshmallowField
 
 from dataclasses_json.undefined import Undefined, UndefinedParameterError
 
+ExcludePredicate = Callable[[str, Any], bool]
+
+
+class _Exclude(NamedTuple):
```

Comment:
also commented [here](https://github.com/lidatong/dataclasses-json/pull/221#issuecomment-633634103) but why a `NamedTuple`?

Reply:
With an enum, the `NEVER` and `ALWAYS` members are of the `Exclude` type. With a namedtuple, they are of the `ExcludePredicate` type. This keeps the type signature of `config(exclude=)` simple, accepting `Optional[ExcludePredicate]` instead of `Optional[Union[Exclude,ExcludePredicate]]`.

Reply:
Got it. What about solving the problem with this pattern?

```python
class Exclude:  #avoids `Enum` or `NamedTuple` altogether
      ALWAYS = lambda _: True
```

I'd favor keeping it as a `Callable` type, rather than introducing a type alias (mainly to avoid adding to the library's API)

Have you considered having the `Callable` signature accept a `TypeVar` instead of `Any`?

This way the type argument is captured and type-checked by users optionally using `mypy` or similar tooling

Reply:
* NamedTuples are immutable, while classes aren't. I don't have strong opinions here, so let me know which sits better and I'll make the change.
* Agreed regarding keeping the API clean, I'll remove the alias.
* I think mypy is capable of narrowing `Any` based on what it sees at analysis-time, with a TypeVar only being useful to enable overloading, but using the TypeVar won't hurt things (and looks nicer), so I'm fine with that. I'll make the change and push it up :+1:

Reply:
> It's just a static constant that scoped within a class (stylistically), could be module-level

We do get some niceties out of the box - attributes cannot be overwritten (raises an AttributeError), and new attributes cannot be added at runtime.  With Python 3.8, we do get closer with the `Final` type annotation, but that's not enforced at runtime.

Reply:
Understand your point, but I think a namedtuple in which only a single instance is ever intended to exist adds confusion and obscures -- I'd rather achieve this functionality of "runtime enforcement" by overriding the dunder methods. But I'm going to make a call to simplify here and just have an uppercase constant that is constant by convention